src/poetry/installation/chef.py

```diff
@@ -81,7 +81,19 @@ def get_cached_archives_for_link(self, link: Link) -> list[Link]:
         links = []
         for archive_type in archive_types:
             for archive in cache_dir.glob(f"*.{archive_type}"):
-                links.append(Link(archive.as_uri()))
+                accept = True
+                if link.hash is not None:
+                    real_hash = get_file_hash(
+                        archive, link.hash_name  # type: ignore[arg-type]
+                    )
+                    if real_hash != link.hash:
+                        _logger.warning(
+                            f"Cache of {link.filename} is corrupted. The file will be"
+                            " reloaded."
```

Comment:
Perhaps including the hashes in the error messages might help people spot weird problems?

i.e., if a bunch of cached files were truncated to 0 bytes, they'd all log with the same hash, and users would know something wacky had happened.

Or it might be pointless noise.  Who knows?